### PR TITLE
Parser refactor & fixups

### PR DESCRIPTION
**Pull request by GeorgeSittas**

### CODE REVIEW

**Comment by GeorgeSittas** [comment]
Main points of interest for this PR are the following helper methods introduced to the parser:

```python
    def _parse_wrapped_id_vars(self):
        return self._parse_wrapped_csv(lambda: self._parse_id_var())

    def _parse_wrapped_csv(self, parse_method, sep=TokenType.COMMA):
        return self._parse_wrapped(lambda: self._parse_csv(parse_method, sep=sep))

    def _parse_wrapped(self, parse_method):
        self._match_l_paren()
        parse_result = parse_method()
        self._match_r_paren()
        return parse_result
```

These seemed useful for simplifying some parts where we repeated the `match_l_paren, parse, match_r_paren` pattern.

File: sqlglot/parser.py [hunk 1]
```diff
@@ -893,7 +894,7 @@ def _parse_auto_increment(self):
         return self.expression(
             exp.AutoIncrementProperty,
             this=exp.Literal.string("AUTO_INCREMENT"),
-            value=self._parse_var() or self._parse_number(),
+            value=self._parse_number(),
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:897]: This was changed because I couldn't find any documented examples where parsing a VAR made sense. Probably needs to be double-checked, though.

File: sqlglot/parser.py [hunk 2]
```diff
@@ -850,7 +852,6 @@ def _parse_partitioned_by(self):
 
     def _parse_stored(self):
         self._match(TokenType.ALIAS)
-        self._match(TokenType.EQ)
```
Comment (tobymao) [inline sqlglot/parser.py:853]: why was this removed? i assume it's so that you can do stored="parquet" or something

File: sqlglot/parser.py [hunk 3]
```diff
@@ -949,33 +950,29 @@ def _parse_properties(self):
 
         while True:
             if self._match(TokenType.WITH):
-                self._match_l_paren()
-                properties.extend(self._parse_csv(lambda: self._parse_property()))
-                self._match_r_paren()
+                properties.extend(self._parse_wrapped_csv(lambda: self._parse_property()))
```
Comment (tobymao) [inline sqlglot/parser.py:953]: parse_wrapped_csv(self._parse_property)

File: sqlglot/parser.py [hunk 4]
```diff
@@ -1078,12 +1075,7 @@ def _parse_cache(self):
         options = []
 
         if self._match(TokenType.OPTIONS):
-            self._match_l_paren()
-            k = self._parse_string()
-            self._match(TokenType.EQ)
-            v = self._parse_string()
-            options = [k, v]
-            self._match_r_paren()
+            options = self._parse_wrapped_csv(lambda: self._parse_string(), sep=TokenType.EQ)
```
Comment (tobymao) [inline sqlglot/parser.py:1078]: csv(self._parse_string)

File: sqlglot/parser.py [hunk 5]
```diff
@@ -1099,27 +1091,13 @@ def _parse_partition(self):
...
-
-            if self._match(TokenType.EQ):
-                value = self._parse_string()
-
-            return exp.Property(this=key, value=value)
-
-        self._match_l_paren()
-        values = self._parse_csv(parse_values)
-        self._match_r_paren()
+            props = self._parse_csv(lambda: self._parse_var_or_string(), sep=TokenType.EQ)
```
Comment (tobymao) [inline sqlglot/parser.py:1094]: same

File: sqlglot/parser.py [hunk 6]
```diff
@@ -1099,27 +1091,13 @@ def _parse_partition(self):
...
-            exp.Partition,
-            this=values,
-        )
+        return self.expression(exp.Partition, this=self._parse_wrapped_csv(lambda: parse_values()))
 
     def _parse_value(self):
-        self._match_l_paren()
-        expressions = self._parse_csv(self._parse_conjunction)
-        self._match_r_paren()
+        expressions = self._parse_wrapped_csv(lambda: self._parse_conjunction())
```
Comment (tobymao) [inline sqlglot/parser.py:1100]: same

File: sqlglot/parser.py [hunk 7]
```diff
@@ -1213,13 +1186,9 @@ def _parse_cte(self):
...
             self.raise_error("Expected AS in CTE")
 
-        self._match_l_paren()
-        expression = self._parse_statement()
-        self._match_r_paren()
-
         return self.expression(
             exp.CTE,
-            this=expression,
+            this=self._parse_wrapped(lambda: self._parse_statement()),
```
Comment (tobymao) [inline sqlglot/parser.py:1191]: same

File: sqlglot/parser.py [hunk 8]
```diff
@@ -1437,12 +1398,8 @@ def _parse_unnest(self):
         if not self._match(TokenType.UNNEST):
             return None
 
-        self._match_l_paren()
-        expressions = self._parse_csv(self._parse_column)
-        self._match_r_paren()
-
+        expressions = self._parse_wrapped_csv(lambda: self._parse_column())
```
Comment (tobymao) [inline sqlglot/parser.py:1401]: same

File: sqlglot/parser.py [hunk 9]
```diff
@@ -1589,11 +1530,7 @@ def _parse_group(self, skip_group_by_token=False):
     def _parse_grouping_sets(self):
         if not self._match(TokenType.GROUPING_SETS):
             return None
-
-        self._match_l_paren()
-        grouping_sets = self._parse_csv(self._parse_grouping_set)
-        self._match_r_paren()
-        return grouping_sets
+        return self._parse_wrapped_csv(lambda: self._parse_grouping_set())
```
Comment (tobymao) [inline sqlglot/parser.py:1533]: same

File: sqlglot/parser.py [hunk 10]
```diff
@@ -2414,20 +2327,14 @@ def _parse_trim(self):
 
     def _parse_window(self, this, alias=False):
         if self._match(TokenType.FILTER):
-            self._match_l_paren()
-            this = self.expression(exp.Filter, this=this, expression=self._parse_where())
-            self._match_r_paren()
+            where = self._parse_wrapped(lambda: self._parse_where())
```
Comment (tobymao) [inline sqlglot/parser.py:2330]: same

File: sqlglot/parser.py [hunk 11]
```diff
@@ -2414,20 +2327,14 @@ def _parse_trim(self):
...
         # https://learn.microsoft.com/en-us/sql/t-sql/functions/percentile-disc-transact-sql?view=sql-server-ver16
         if self._match(TokenType.WITHIN_GROUP):
-            self._match_l_paren()
-            this = self.expression(
-                exp.WithinGroup,
-                this=this,
-                expression=self._parse_order(),
-            )
-            self._match_r_paren()
+            order = self._parse_wrapped(lambda: self._parse_order())
```
Comment (tobymao) [inline sqlglot/parser.py:2336]: same

File: sqlglot/parser.py [hunk 12]
```diff
@@ -2634,17 +2530,20 @@ def _parse_tokens(self, parse_method, expressions):
         return this
 
     def _parse_wrapped_id_vars(self):
+        return self._parse_wrapped_csv(lambda: self._parse_id_var())
```
Comment (tobymao) [inline sqlglot/parser.py:2533]: same

**Review by tobymao — APPROVED** [review]
